src/Provider/MapQuest/GeocodeQuery.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\MapQuest;
+
+use Geocoder\Formatter\StringFormatter;
+use Geocoder\Location;
+use Geocoder\Query\GeocodeQuery as BaseGeocodeQuery;
+
+class GeocodeQuery extends BaseGeocodeQuery implements GetAddressInterface
+{
+    const DATA_KEY_ADDRESS = 'address';
+
+    protected function __construct($address)
+    {
+        if ($address instanceof Location) {
+            $this->data[static::DATA_KEY_ADDRESS] = $address;
+        } else {
+            parent::__construct($address);
+        }
+    }
+
+    public static function createFromAddress(Location $address): self
+    {
+        return new self($address);
```

Comment:
Just do: 

```php
$foo = new self('');
return $foo->withData(static::DATA_KEY_ADDRESS], $address);
```

Reply:
It is a bit more ineffecient, as it will create two objects instead of one, and then leave the first one to be garbage collected.
But yes, that is a valid way to do it if the methods in the base class cannot be changed to be protected instead of private.

Reply:
It is really not that inefficient. We are talking optimization on less than duoble quotes vs single quotes.

Reply:
Ok, forget my inefficency arguments then. Please respond to the other points I raised.